### PR TITLE
WIP: Create endpoit for moduleTypes drop down menu ticket5

### PR DESCRIPTION
**Pull request by Shimaadnan**
Create endpoit for moduleTypes drop down menu

### CODE REVIEW

File: client/src/pages/Admin.js [hunk 1]
```diff
@@ -51,43 +60,94 @@ function AdminPage() {
...
+  } catch (error) {
+    console.error(error);
+    return "No Module is not avaialbe";
+  }
+}
+
+
+useEffect(() => {
+	getAvailableModules().then((modulesResult) => setModules(modulesResult));
+}, [selectedModuleType]);
```
Comment (salehpirouzi-cyf) [inline client/src/pages/Admin.js:77]: if we don't expect the list of module types to be changed very often, you can move this to be done only when we load the page (in the other useEffect)

File: client/src/pages/Admin.js [hunk 2]
```diff
@@ -51,43 +60,94 @@ function AdminPage() {
...
+	default:
+		formComponent = null;
+}
+	
 
-	return (
+  
+	if (!pagesData || !modules) <p>Loading..</p>;
+	
+	console.log(selectedModuleType, selectedPage);
```
Comment (salehpirouzi-cyf) [inline client/src/pages/Admin.js:112]: console.log!!

File: server/api.js [hunk 3]
```diff
@@ -69,20 +73,170 @@ router.delete("/pages/:page_title/:record_id", async (req, res) => {
...
+	const { boldText, normalText, background } = req.body;
+
+	if (!boldText || !normalText || !background) {
+		res.status(500).json("Please fill all the fields");
+	}
+
+	try {
+
+		// First inserting the textBanner table
+		const textBannerData = await db.query(
```
[the remark this answers is not in the document]
Comment (khmdagal) [reply server/api.js:112]: Done

File: client/src/pages/Admin.js [hunk 4]
```diff
@@ -51,43 +60,94 @@ function AdminPage() {
...
+    return modulesData;
+  } catch (error) {
+    console.error(error);
+    return "No Module is not avaialbe";
+  }
+}
+
+
+useEffect(() => {
+	getAvailableModules().then((modulesResult) => setModules(modulesResult));
```
Comment (salehpirouzi-cyf) [inline client/src/pages/Admin.js:76]: Why don't you put this in the other useEffect that you have?

File: server/api.js [hunk 5]
```diff
@@ -69,20 +74,183 @@ router.delete("/pages/:page_title/:record_id", async (req, res) => {
...
+		);
+		const record_id = lastRecord.rows[0].record_id;
+
+		// fourth inserting the above datas into the modules table
+		const insertingModulesTable = await db.query(
+			`insert into modules(page_id,module_type,record_id) values($1,$2,$3)`,
+			[page_id, "heroBanner", record_id]
+		);
+		await db.query("commit");
+		return res.status(200).json({
```
Comment (salehpirouzi-cyf) [inline server/api.js:245]: if you return here, it never gets to 'finally' block which means the db is never released

Comment (khmdagal) [reply server/api.js:241]: Remove all the db transaction bloks

File: client/src/pages/Admin.js (hunk 4 above)
Comment (khmdagal) [reply client/src/pages/Admin.js:76]: Delete one useEffect and only one useEffect is in use now.

**Pull request merged**